### PR TITLE
#989 auto focusing item input

### LINKED ISSUE #989: auto focus autocomplete when adding a line
When clicking add a line (or when clicking "next" on an edit modal") the auto complete should be auto focused. It is making me furious!!

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #989 

- Just wanted to add a little auto focus... and then..
- I added `openOnFocus` to the autocomplete which was kind of nice, but I was having a little bit of an issue where the list box of the auto complete was not aligning well after a transition animation. Weird. So I left that out and just went with the autoFocus (If you want to see, add `openOnFocus={true}` to the autocomplete..
- I realised the next behaviour was a little inconsistent in all the modals. Some were disabling while the modal was in Create mode. But I wanted it enabled! Especially with the auto focus it makes it really easy to add items with the next button. Though, it does make things a little inconsistent still - the next behaviour during 'update' mode - disabling the next button? Closing the modal when there are no more 'nexts'? Go to 'create mode' when there are no more nexts? 🤷 🤷 🤷 

Some further improvements for a bit better UX
- Add shortcuts to stocktake tabs
- Add shortcuts to the OK/OK&Next buttons/ Cancel has Esc already
- Add shortcut to the "add x" button
- Make the tabbing stay within the table once you're there?
- Even though the UI for the requisition lines probably looks better with the comment in the middle, the tabbing is no ideal and might be a better experience to have the comment be on the right 🤔

### CODE REVIEW

File: packages/common/src/hooks/useDialog/useDialog.tsx [hunk 1]
```diff
@@ -128,7 +128,7 @@ export const useDialog = (dialogProps?: DialogProps): DialogState => {
           sx={{ overflowX: 'hidden', ...contentSX }}
         >
           <Slide in={slideConfig.in} direction={slideConfig.direction}>
-            <div> {children}</div>
+            <div>{slideConfig.in && children}</div>
```
Comment (joshxg) [inline packages/common/src/hooks/useDialog/useDialog.tsx:131]: Making whatever is sliding unmount (triggers auto focus 🤔 )

It does actually unmount regularly, but will remount just before the animation begins

File: packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx [hunk 2]
```diff
@@ -44,7 +44,8 @@ export const StocktakeLineEdit: FC<StocktakeLineEditProps> = ({
 
   const onNext = async () => {
     await save(draftLines);
-    if (nextItem) setCurrentItem(nextItem);
+    if (mode === ModalMode.Update && nextItem) setCurrentItem(nextItem);
+    else if (mode === ModalMode.Create) setCurrentItem(null);
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx:48]: umm.. not sure but made it the same at least?

**Comment by mark-prins** [comment]
oh, that's hilarious! I made the same changes this morning.. having forgotten about that issue you raised yesterday!!

Super annoying, not being auto focussed 😁

File: packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx (hunk 2 above)
Comment (mark-prins) [reply packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEdit.tsx:48]: 👍  very nice user experience!

File: packages/common/src/hooks/useDialog/useDialog.tsx (hunk 1 above)
Comment (mark-prins) [reply packages/common/src/hooks/useDialog/useDialog.tsx:131]: thanks - I wondered about that. was noticeable in request reqs

**Review by mark-prins — APPROVED** [review]
😁  awesome! nice use of offroads too!

**Pull request merged**